### PR TITLE
Clarify that axis can be None in nn functions log_softmax, softmax, and standardize.

### PR DESCRIPTION
**Pull request by carlosgmartin**
Clarify that `axis` can be `None` in the following functions:

- [`log_softmax`](https://docs.jax.dev/en/latest/_autosummary/jax.nn.log_softmax.html)
- [`softmax`](https://docs.jax.dev/en/latest/_autosummary/jax.nn.softmax.html)
- [`standardize`](https://docs.jax.dev/en/latest/_autosummary/jax.nn.standardize.html)

### CODE REVIEW

File: jax/_src/nn/functions.py [hunk 1]
```diff
@@ -668,7 +668,7 @@ def standardize(x: ArrayLike,
 
   Args:
     x: input array to be standardized.
-    axis: integer or tuple of integers representing the axes along which
+    axis: integer, tuple of integers, or ``None`` (all axes), representing the axes along which
```
[the remark this answers is not in the document]
Comment (carlosgmartin) [reply jax/_src/nn/functions.py]: Done.

File: jax/_src/nn/functions.py [hunk 2]
```diff
@@ -591,7 +591,7 @@ def softmax(x: ArrayLike,
     x : input array
     axis: the axis or axes along which the softmax should be computed. The
       softmax output summed across these dimensions should sum to :math:`1`.
-      Either an integer or a tuple of integers.
+      Either an integer, tuple of integers, or ``None`` (all axes).
```
Comment (jakevdp) [inline jax/_src/nn/functions.py:594]: Can you make the same change for `jax.scipy.special.softmax` and `log_softmax`?

**Review by jakevdp — COMMENTED** [review]
Thanks! One quick request below:

File: jax/_src/nn/functions.py (hunk 2 above)
Comment (carlosgmartin) [reply jax/_src/nn/functions.py:594]: Done.

**Review by jakevdp — APPROVED** [review]
Thanks!
